packages/financial-templates-lib/src/price-feed/CreatePriceFeed.js

```diff
@@ -5,12 +5,12 @@ const { CryptoWatchPriceFeed } = require("./CryptoWatchPriceFeed");
 const { UniswapPriceFeed } = require("./UniswapPriceFeed");
 const { BalancerPriceFeed } = require("./BalancerPriceFeed");
 const { defaultConfigs } = require("./DefaultPriceFeedConfigs");
-
-const Uniswap = require("@uma/core/build/contracts/Uniswap.json");
-const ExpiringMultiParty = require("@uma/core/build/contracts/ExpiringMultiParty.json");
-const Balancer = require("@uma/core/build/contracts/Balancer.json");
+const { getTruffleContract } = require("@uma/core");
 
 async function createPriceFeed(logger, web3, networker, getTime, config) {
+  const Uniswap = getTruffleContract("Uniswap", web3, "latest");
+  const Balancer = getTruffleContract("Balancer", web3, "latest");
```

Comment:
small nit here, wont block pr with this, but the tests offer a way to change contract version, though its hard coded here. does this cause a problem if tests specify something other than latest? its also fine if latest is always to be assumed, that could be reasonable.

Reply:
> does this cause a problem if tests specify something other than latest?

Yes this does cause a problem if the tests specify a different version, this is why I think we [should explore exporting the "valid" versions from the client to the tests](https://github.com/UMAprotocol/protocol/pull/2344#discussion_r549693664), so that the tests are only testing the expected versions.  Do you think thats a good approach?

Reply:
im not sure actually... my gut instinct is i probably wouldn't want to do it that way, but i could be missing something since this is a broader convo. I feel like we should treat tests, software and contracts as part of the same versioned package. That way as contracts evolves, tests and supporting software can evolve with it without worry of breaking tests for some other version of the software.  Probably too little too late for something like this and you probably have better insight as to what a workable solution is at this point, so i would defer to your judgement here.

Ideally though it would be something that doesn't add increasing maintenance cost as the number of versions go up.

Reply:
> I feel like we should treat tests, software and contracts as part of the same versioned package. That way as contracts evolves, tests and supporting software can evolve with it without worry of breaking tests for some other version of the software.

This I 100% agree with. I guess herein lies the problem:
- `liquidator` package imports `@uma/core:1.2.2` and uses that version import a contract ABI to create a new `ExpiringMultiPartyClient`
- `ExpiringMultiPartyClient` technically does not restrict which ABI version can be imported, but it will break if an ABI > `1.2.2` is imported. Accordingly, we explicitly set the contract ABI's to `1.2.2` in the `ExpiringMultiPartyClient` tests.

After writing this out, one solution would be to assert within the `ExpiringMultiPartyClient` that the inputted ABI is the correct version. I'm not sure there is a way to check an ABI "version" (in the @uma/core sense of the word), but I think within the client is the logical place to restrict versions.

Thoughts @UMAprotocol/eng ?